rfcs/0000-lifecycle-crd.md

```diff
@@ -0,0 +1,137 @@
+`[Readable](https://github.com/your-name-or-org/kpack/blob/<pr-branch-name>/rfcs/0000-<my-feature>.md)`
+
+**Problem:**
+
+The CNB lifecycle is provided to kpack as in image reference in a configMap similar to how the completion and prepare
+images are passed to the kpack controller. This means that every builder will always use the same exact lifecycle, with
+no ability to specify an alternate one that may support different buildpack api or platform api versions. Due to kpack
+support of windows, a custom lifecycle image with specific layers for the linux and windows binaries is required
+(with a label specifying the layer diff id for each os). Because of this, a cve present in the lifecycle that has been 
+patched upstream, requires any user to build the kpack specific lifecycle image themselves if the kpack provided one
+has not been bumped.
+
+As we [move to deprecate windows](https://github.com/buildpacks-community/kpack/discussions/1366), there is an opportunity for us to utilize
```

Comment:
We do ship "lifecycle images" for Windows, so in theory you could specify the Windows image if Windows support is desired.

Reply:
Do we want to support multi-arch lifecycle images? I.e. if I have an Image Index with linux/amd64, linux/arm64, and windows/amd64 images, do I need to break them out into 3 separate ClusterLifecycles?

Reply:
That is a good question. I suspect the simpler approach would be to keep them separate, but I could be wrong. Would appreciate other more knowledgeable folks weighing in here.